internal/util/tls/tls.go

```diff
@@ -0,0 +1,48 @@
+package tls
+
+import (
+	"crypto/tls"
+	"fmt"
+	"os"
+)
+
+// ExtractClientCertificates extracts tls.Certificates from TLSClientConfig.
+// It returns an empty slice in case there was no client cert and/or client key provided.
+func ExtractClientCertificates(cert, certFile, key, keyFile string) ([]tls.Certificate, error) {
```

Comment:
How about we accept here `[]byte`s as types for `cert` and `key`? This way we give a hint to the caller instead of accepting 4 string params which might be easy to trip over.

Reply:
done.